modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoveryNetworkIssuesTest.java

```diff
@@ -383,6 +415,54 @@ private void simulateFailureOfTwoNodes(boolean sequentionally) throws Exception
         }
     }
 
+    /**
+     * This test uses node failure by stopping service threads, which makes the node unresponsive and results in
+     * failing connection to the server.
+     *
+     * @throws Exception If failed.
+     * @see TcpDiscoverySpi#simulateNodeFailure()
+     */
+    @Test
+    public void testCheckNodeFailureSocketConnectionLogMessage() throws Exception {
+        ListeningTestLogger testLog = new ListeningTestLogger(log);
+
+        Collection<LogListener> lsnrs = new ArrayList<>();
+
+        lsnrs.add(LogListener.matches("Checking connection to node").andMatches("result=failed").times(1).build());
+        lsnrs.add(LogListener.matches("Connection check to previous node failed").times(1).build());
+
+        lsnrs.forEach(testLog::registerListener);
+
+        TcpDiscoverySpi spi0 = new TcpDiscoverySpi();
+
+        startGrid(getTestConfigWithSpi(spi0, "ignite-0"));
+
+        IgniteConfiguration cfg1 = getTestConfigWithSpi(new TcpDiscoverySpi(), "ignite-1");
+        cfg1.setGridLogger(testLog);
+
+        startGrid(cfg1);
+
+        startGrid(getTestConfigWithSpi(new TcpDiscoverySpi(), "ignite-2"));
+
+        spi0.simulateNodeFailure();
+
+        for (LogListener lsnr : lsnrs)
+            waitForCondition(lsnr::check, getTestTimeout());
```

Comment:
Let's ensure node 0 correctly gone but everything else is OK. Like this:

`waitForCondition(() -> grid(1).cluster().nodes().size() == 2, getTestTimeout());
waitForCondition(() -> grid(2).cluster().nodes().size() == 2, getTestTimeout());

assertTrue(F.viewReadOnly(grid(1).cluster().nodes(), n -> n, n -> n.order() == 1).isEmpty());
assertTrue(F.viewReadOnly(grid(2).cluster().nodes(), n -> n, n -> n.order() == 1).isEmpty());`